types/sensative/strip/uplink.js

```diff
@@ -0,0 +1,211 @@
+        target.temperature = ((bytes[pos] & 0x80 ? 0xFFFF << 16 : 0) | (bytes[pos++] << 8) | bytes[pos++]) / 10;
+        break;
+      case 112: // Capacitance Raw Sensor Value 2bytes 0-65535
+        target.capacitanceFlood = ((bytes[pos++] << 8) | bytes[pos++]); // should never trigger anymore
+        break;
+      case 113: // Capacitance Raw Sensor Value 2bytes 0-65535
+        target.capacitancePad = ((bytes[pos++] << 8) | bytes[pos++]); // should never trigger anymore
+        break;
+      case 110:
+        pos += 8;
+        break;
+      case 114: // Capacitance Raw Sensor Value 2bytes 0-65535
+        target.capacitanceEnd = ((bytes[pos++] << 8) | bytes[pos++]); // should never trigger anymore
+        break;
+    }
+  }
+
+  var decoded = {};
+  var pos = 0;
+  var type;
+
+  switch (port) {
+    case 1:
+      if (bytes.length < 2) {
+        decoded.error = 'Wrong length of RX package';
+        break;
+      }
+      decoded.historySeqNr = (bytes[pos++] << 8) | bytes[pos++];
+      decoded.prevHistSeqNr = decoded.historySeqNr;
+      while (pos < bytes.length) {
+        type = bytes[pos++];
+        if (type & 0x80)
+          decoded.prevHistSeqNr--;
+        decodeFrame(type, decoded);
+      }
+      break;
+
+    case 2:
+      var now = new Date();
+      decoded.history = {};
+      if (bytes.length < 2) {
+        decoded.history.error = 'Wrong length of RX package';
+        break;
+      }
+      var seqNr = (bytes[pos++] << 8) | bytes[pos++];
+      while (pos < bytes.length) {
+        decoded.history[seqNr] = {};
+        decoded.history.now = now.toUTCString();
+        secondsAgo = (bytes[pos++] << 24) | (bytes[pos++] << 16) | (bytes[pos++] << 8) | bytes[pos++];
+        decoded.history[seqNr].timeStamp = new Date(now.getTime() - secondsAgo * 1000).toUTCString();
+        type = bytes[pos++];
+        decodeFrame(type, decoded.history[seqNr]);
+        seqNr++;
+      }
+  }
+  return decoded;
+}
+
+function parseHexString(str) {
+  var result = [];
+  while (str.length >= 2) {
+    result.push(parseInt(str.substring(0, 2), 16));
+
+    str = str.substring(2, str.length);
+  }
+
+  return result;
+}
+
+function consume(event) {
+  var payload = event.data.payload_hex;
+  var port = event.data.port;
+  var data = decoder(parseHexString(payload), port);
+
+  var lifecycle = {}; var alarm = {}; var def = {};
+  var lifecycleOut = false; var alarmOut = false; var defOut = false;
+  var defaultKeys = ["temperature", "averageTemperature", "humidity", "lux", "lux2", "open", "tamperReport", "flood", "doorCount", "presence", "irProximity", "irCloseproximity"];
+  var alarmKeys = ["highAlarm", "lowAlarm", "doorAlarm", "tamperAlarm", "floodAlarm", "foilAlarm", "userSwitch1Alarm", "closeProximityAlarm", "disinfectAlarm"];
+  var lifecycleKeys = ["statusPercent", "error", "historySeqNr", "prevHistSeqNr"];
+
+  Object.keys(data).forEach(function (key) {
```

Comment:
not sure if this is a good way of handling that, better make it explicit

Reply:
do we want to handle it the same as decentlab ?